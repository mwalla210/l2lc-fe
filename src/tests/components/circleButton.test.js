import React from 'react'
import renderer from 'react-test-renderer'
import CircleButton from '../../components/circleButton'

jest.mock('reactstrap', () => ({
  Button: 'Button'
}))

const defaultOptions = {
  onClick: jest.fn(),
  iconName: 'iconName',
  styleProps: {},
  disabled: false
}

describe('ButtonDefault', () => {
  it ('Renders with snapshot (disabled: false)', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <CircleButton {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (disabled: true)', () => {
    let options = Object.assign({}, defaultOptions)
    options.disabled = true
    const component = renderer.create(
      <CircleButton {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
