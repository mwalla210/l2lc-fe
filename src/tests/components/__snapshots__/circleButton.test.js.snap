// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDefault Renders with snapshot (disabled: false) 1`] = `
<Button
  aria-label="Left Align"
  className="btn-circle"
  color="secondary"
  disabled={false}
  onClick={[MockFunction]}
  outline={true}
>
  <img
    alt="iconName"
    src="../../style/open-iconic-master/svg/iconName.svg"
    style={null}
  />
</Button>
`;

exports[`ButtonDefault Renders with snapshot (disabled: true) 1`] = `
<Button
  aria-label="Left Align"
  className="btn-circle"
  color="secondary"
  disabled={true}
  onClick={[MockFunction]}
  outline={true}
>
  <img
    alt="iconName"
    src="../../style/open-iconic-master/svg/iconName.svg"
    style={null}
  />
</Button>
`;
